### PR TITLE
Replace deprecated CircleCI image

### PR DESCRIPTION
**Pull request by dragonsahead**
Fixes https://github.com/metabase/metabase/issues/18650

EDIT 04-16-2022: upgraded Clojure CLI version + made the builder image multi-arch

Tests on https://app.circleci.com/pipelines/github/metabase/metabase?branch=check-new-builder&filter=all

### CODE REVIEW

**Comment by paoliniluis** [comment]
I'm next to the finish line:
1) image is built from new Temurin JDK + Node LTS + Clojure (8, 11 and 17)
2) user profile is configured as circleci wants, so it's plug and play with the current CI
3) it's slimmer (only ~600mb compared to current circleci image)
4) it's very very upgradable (in fact I'm thinking into building one to build and other to test, so it's not a full image for every step)

but I can't get yarn to spawn a java process... but I'm sure it's a path thing somewhere that I'll have to fix

**Comment by paoliniluis** [comment]
"be-tests-ee" and "be-tests-oss" are pending since those were changed "be-tests-java-8-ee" "be-tests-java-8-oss" to know which java version we're running those tests

**Review by camsaul — APPROVED** [review]

**Comment by paoliniluis** [comment]
Now this should also be ready to merge once all tests pass

File: .circleci/config.yml
Comment (diogormendes) [inline .circleci/config.yml:1084]: No need to test with 16 anymore ??

Comment (ariya) [reply .circleci/config.yml:1084]: Only 8, 11, and 17 are LTS versions.

**Review by diogormendes — APPROVED** [review]

File: Dockerfile
Comment (diogormendes) [inline Dockerfile:22]: 🚀

**Review by diogormendes — APPROVED** [review]

**Comment by paoliniluis** [comment]
Thanks @diogormendes for the approval, I'll need https://github.com/metabase/metabase-docker-ci/pull/6 approved before merging this one

**Comment by paoliniluis** [comment]
Thanks @diogormendes for the approval, I'll need https://github.com/metabase/metabase-docker-ci/pull/6 approved before merging this one

**Review by camsaul — CHANGES_REQUESTED** [review]
We shouldn't test with Clojure 1.11.0 since we're not shipping anything with it yet.

We'll probably switch to Clojure 1.11.1 when that comes out but we're still shipping 1.10.x until then

**Comment by dpsutton** [comment]
My reading of this is its just bumping the version of the clojure cli. We specify `org.clojure/clojure {:mvn/version "1.10.3"}` so we will not be testing on clojure 1.11.X at all. But we for sure want to stay away from 1.11.0 and only use 1.11.1 and above when we do switch.

**Comment by paoliniluis** [comment]
Thanks @dpsutton and @camsaul, I left my comments here: https://github.com/metabase/metabase-docker-ci/pull/6#issuecomment-1101692558

**Review by camsaul — APPROVED** [review]
Oops you're totally right @dpsutton. Ok in that case then this looks fine

**Pull request merged**
